libsplinter/src/admin/store/diesel/migrations/postgres/migrations/2020-08-18-213009_admin_service_store/up.sql

```diff
@@ -0,0 +1,136 @@
+--
+-- Licensed under the Apache License, Version 2.0 (the "License");
+-- you may not use this file except in compliance with the License.
+-- You may obtain a copy of the License at
+--
+--     http://www.apache.org/licenses/LICENSE-2.0
+--
+-- Unless required by applicable law or agreed to in writing, software
+-- distributed under the License is distributed on an "AS IS" BASIS,
+-- WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+-- See the License for the specific language governing permissions and
+-- limitations under the License.
+-- -----------------------------------------------------------------------------
+
+CREATE TABLE IF NOT EXISTS proposed_circuit (
+    circuit_id                TEXT NOT NULL,
+    authorization_type        TEXT NOT NULL,
+    persistence               TEXT NOT NULL,
+    durability                TEXT NOT NULL,
+    routes                    TEXT NOT NULL,
+    circuit_management_type   TEXT NOT NULL,
+    application_metadata      BINARY NOT NULL,
+    comments                  TEXT NOT NULL,
+    PRIMARY KEY (circuit_id),
+    FOREIGN KEY (circuit_id) REFERENCES proposed_circuit(circuit_id) ON DELETE CASCADE
+);
+
+CREATE TABLE IF NOT EXISTS proposed_node (
+    circuit_id                TEXT NOT NULL,
+    node_id                   TEXT NOT NULL,
+    PRIMARY KEY (circuit_id, node_id),
+    FOREIGN KEY (circuit_id) REFERENCES proposed_circuit(circuit_id) ON DELETE CASCADE
+);
+
+CREATE TABLE IF NOT EXISTS proposed_node_endpoints (
+    circuit_id             TEXT NOT NULL,
+    node_id                TEXT NOT NULL,
+    endpoint               TEXT NOT NULL,
+    PRIMARY KEY (circuit_id, node_id, endpoint),
+    FOREIGN KEY (node_id) REFERENCES proposed_node(node_id),
+    FOREIGN KEY (circuit_id) REFERENCES proposed_circuit(circuit_id) ON DELETE CASCADE
+);
+
+CREATE TABLE IF NOT EXISTS proposed_services (
+    circuit_id                TEXT NOT NULL,
+    service_id                TEXT NOT NULL,
+    service_type              TEXT NOT NULL,
+    PRIMARY KEY (circuit_id, service_id),
+    FOREIGN KEY (circuit_id) REFERENCES proposed_circuit(circuit_id) ON DELETE CASCADE
+);
+
+CREATE TABLE IF NOT EXISTS proposed_services_allowed_nodes (
+    circuit_id                TEXT NOT NULL,
+    service_id                TEXT NOT NULL,
+    allowed_node              TEXT NOT NULL,
+    PRIMARY KEY (circuit_id, service_id, allowed_node),
+    FOREIGN KEY (service_id) REFERENCES proposed_services(service_id),
+    FOREIGN KEY (circuit_id) REFERENCES proposed_circuit(circuit_id) ON DELETE CASCADE
+);
+
+CREATE TABLE IF NOT EXISTS proposed_services_arguments (
+    circuit_id                TEXT NOT NULL,
+    service_id                TEXT NOT NULL,
+    key                       TEXT NOT NULL,
+    value                     TEXT NOT NULL,
+    PRIMARY KEY (circuit_id, service_id),
+    FOREIGN KEY (service_id) REFERENCES proposed_services(service_id) ON DELETE CASCADE
```

Comment:
Looks like you're missing the circuit_id relationship here (which should be the one with the delete cascade)